### PR TITLE
Additional test-program cleanup

### PR DESCRIPTION
**Pull request by MikeOpenHWGroup**
## Description
The goal of this PR is to clean-up some of the "custom" assembly (*.S) test-programs and ensure they can execute properly on the CV32E20.  The PR also addresses some of the feedback from #66.

## Verification
All "custom" C and S test-programs were run on the "core" testbench with Verilator v5.048 and on the "uvmt" testbench with Questasim.

### CODE REVIEW

File: tests/programs/custom/debug_test_reset/debug_test_reset.c [hunk 1]
```diff
@@ -23,6 +23,8 @@
 #include <stdio.h>
```
Comment (cairo-caplan) [inline tests/programs/custom/debug_test_reset/debug_test_reset.c:23]: This tests fail on Questa UVM due to the following assertion being raised:
`CVE2DontSkipExceptionReq`

https://github.com/openhwgroup/cve2/blob/a24bbd2a7a29b5a3062b4865dfb47181d6cc5ab9/rtl/cve2_controller.sv#L829

Did not investiage further just yet

**Review by cairo-caplan — APPROVED** [review]
Hi Mike,

This is great! It looks good to me. I have tested the changes locally too.

File: tests/programs/custom/debug_test_reset/debug_test_reset.c (hunk 1 above)
Comment (MikeOpenHWGroup) [reply tests/programs/custom/debug_test_reset/debug_test_reset.c:23]: Yes, this is one of the [Parked Tests](https://github.com/MikeOpenHWGroup/cv32e20-dv/tree/test-program-cleanup/tests/programs/custom#results-summary), which is a fancy way of saying, "its broken and I don't know why".  The run_c_tests script does not run parked tests.  We'll need to investigate these individually.

`debug_test_reset` is an edge-case test-program that attempts to drive a debug_request while the core is in reset.  It may not be applicable for the CV32E20, or it may need to be updated.

**Comment by MikeOpenHWGroup** [comment]
Thanks @cairo-caplan, merging...
